app/src/main/java/com/github/mateo762/myapplication/ui/habits/CreateHabitScreen.kt

```diff
@@ -261,4 +268,54 @@ private fun saveHabit(context:Context, habitName:String, habitDays:List<DayOfWee
 private fun isValidTime(time: String): Boolean {
     val pattern = Regex(pattern = "^(2[0-3]|[01]?[0-9]):([0-5]?[0-9])\$")
     return pattern.matches(time)
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+private fun scheduleHabit(context: Context, habit: HabitEntity) {
+    val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+    val intent = Intent(context, HabitAlarmReceiver::class.java)
+    intent.putExtra("habit_id", habit.id)
+    intent.putExtra("habit_name", habit.name)
+
+    val endTimeParts = habit.endTime.split(":")
+    val hour = endTimeParts[0].toInt()
+    val minute = endTimeParts[1].toInt()
+
+    for (day in habit.days) {
+        val calendar = Calendar.getInstance().apply {
+            timeInMillis = System.currentTimeMillis()
+            set(Calendar.DAY_OF_WEEK, convertDayOfWeekToInt(day))
+            set(Calendar.HOUR_OF_DAY, hour)
+            set(Calendar.MINUTE, minute)
+            set(Calendar.SECOND, 0)
+        }
+
+        // If the alarm time has already passed today, schedule for next week
+        if (calendar.timeInMillis <= System.currentTimeMillis()) {
+            calendar.add(Calendar.DAY_OF_YEAR, 7)
+        }
+
+        val pendingIntent = PendingIntent.getBroadcast(
+            context,
+            UUID.randomUUID().hashCode(),
+            intent,
+            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+        )
+        // Cancel any existing alarm for this habit
```

Comment:
What are the conditions for canceling the existing alarm / when does it happen?

Reply:
Basically you cancel an alarm that might have been already been associated with that habit before